src/client.rs

```diff
@@ -1341,7 +1341,11 @@ impl<T: Read + Write> Connection<T> {
 
     fn run_command(&mut self, untagged_command: &str) -> Result<()> {
         let command = self.create_command(untagged_command);
-        self.write_line(command.into_bytes().as_slice())
+        let result = self.write_line(command.into_bytes().as_slice());
+        if result.is_err() {
+            self.tag -= 1; // rollback tag increase in create_command()
```

Comment:
Following the discussion in https://github.com/jonhoo/rust-imap/issues/229, I'm okay with this change, though I think we should also add a `skip_tag` method to `Client` to allow manually incrementing the tag.